src/rewrite_gelu.cpp

```diff
@@ -53,7 +58,60 @@ struct find_gelu_erf
     }
 };
 
-void rewrite_gelu::apply(module& m) const { match::find_matches(m, find_gelu_erf{}); }
+/**
+ * Find fastGELU blocks (where the graph already does a GELU approximation) and replace them
+ * with an alternative approximation that is less likely to overflow.
+ * The replacement approximation is equivalent to:
+ * GELU(x) ~= 0.5 * x * ( 1 + tanh( sqrt(2/M_PI) * (x + 0.044715 * x^3)))
+ */
+struct find_tanh_fast_gelu
+{
+    auto matcher() const { return match::gelu_tanh(); }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        // Algebraically rearranged version of the tanh GELU approximation that is more numerically
+        // stable
+        auto ins        = r.result;
+        auto x          = r.instructions["x"];
+        auto sqrt_2_rpi = m.add_literal(
+            literal{shape{x->get_shape().type()},
+                    {0.7978845608028653558798921198687637369517172623298693153318516593}});
+        auto fit_const = m.add_literal(literal{shape{x->get_shape().type()}, {0.044715f}});
```

Comment:
What is this value? Can it be computed as well?

Reply:
This is a fitting constant, so it cannot be computed.

Reply:
How is it fitted? Is it using a lagrange polynomial or taylor series?

Reply:
Specified in https://onnx.ai/onnx/operators/onnx__Gelu.html

Reply:
The fitting is based on the original GELU paper by Hendrycks and Gimpel (https://arxiv.org/abs/1606.08415) who reference a paper by Choudhry (https://www.hrpub.org/download/20140305/MS7-13401470.pdf). They probably used some measure of difference between the original erf() version and `0.5(1+tanh(y))` and did a fitting.

Reply:
Can you add a comment in the code where you got the number from